gdscripts/console.gd

```diff
@@ -0,0 +1,391 @@
+extends Node
+
+# Based on https://github.com/jitspoe/godot-console (MIT)
+# some errors fixed, features needs to be added
+
+# TODO:
+# - autocompletion for command arguments
+# - losing input focus
+
+var enabled := true
+var enable_on_release_build := false : set = set_enable_on_release_build
+var pause_enabled := false
+
+signal console_opened
+signal console_closed
+signal console_unknown_command
+
+
+class ConsoleCommand:
+    var function : Callable
+    var arguments : PackedStringArray
+    var required : int
+    var description : String
+    func _init(in_function : Callable, in_arguments : PackedStringArray, in_required : int = 0, in_description : String = ""):
+        function = in_function
+        arguments = in_arguments
+        required = in_required
+        description = in_description
+
+
+@onready var control := Control.new()
+@onready var rich_label := RichTextLabel.new()
+@onready var line_edit := LineEdit.new()
```

Comment:
Can't we just have them in the inspector? Is there a need to instantiate them? (If so, please describe in code comment to make explanation persistent)

Reply:
^^ (about implementation of console.gd)

> Is there a need to instantiate them?

This approach requires no scene (tscn) file. `Console` is registered as singleton and accessible anywhere by `Console` global.